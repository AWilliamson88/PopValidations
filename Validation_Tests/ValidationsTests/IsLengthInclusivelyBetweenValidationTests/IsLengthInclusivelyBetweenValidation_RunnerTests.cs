using ApprovalTests;
using FluentAssertions;
using PopValidations;
using PopValidations_Tests.TestHelpers;
using PopValidations_Tests.ValidationsTests.GenericTestableObjects;
using System.Collections.Generic;
using System.Threading.Tasks;
using Xunit;

namespace PopValidations_Tests.ValidationsTests.IsLengthExclusivelyBetweenValidationTests;

public class IsLengthInclusivelyBetween_NoError_TestingValidator
    : AbstractValidator<NullAllFieldTypesDto>
{
    public IsLengthInclusivelyBetween_NoError_TestingValidator()
    {
        Describe(x => x.String).IsLengthInclusivelyBetween(0, 10);
        Describe(x => x.AllFieldTypesList).IsLengthInclusivelyBetween(0, 1);
        Describe(x => x.AllFieldTypesLinkedList).IsLengthInclusivelyBetween(0, 1);
        Describe(x => x.AllFieldTypesIEnumerable).IsLengthInclusivelyBetween(0, 1);
        Describe(x => x.AllFieldTypesDictionary).IsLengthInclusivelyBetween(0, 1);
    }
}

public class IsLengthInclusivelyBetweenValidation_RunnerTests
{
    [Fact]
    public async Task GivenAValidator_WithNoErrors_ThenNoErrorsAreProduced()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(
            new IsLengthInclusivelyBetween_NoError_TestingValidator()
        );

        // Act
        var validationResult = await runner.Validate(new NullAllFieldTypesDto() { String = "bbb", });

        // Assert
        validationResult.Errors.Should().BeEmpty();
    }

    [Fact]
    public async Task GivenAValidator_WithErrors_ThenEveryFieldHasErrors()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(
            new IsLengthInclusivelyBetween_NoError_TestingValidator()
        );

        NullAllFieldTypesDto dto =
            new()
            {
                String = "zzzzzzzzzzzzzzzzzzzz",
                AllFieldTypesList = new() { new NonNullAllFieldTypesDto(), new NonNullAllFieldTypesDto() },
                AllFieldTypesLinkedList = new(new[] { new NonNullAllFieldTypesDto(), new NonNullAllFieldTypesDto() }),
                AllFieldTypesIEnumerable = new List<NonNullAllFieldTypesDto>()
                {
                    new NonNullAllFieldTypesDto(),
                    new NonNullAllFieldTypesDto()
                },
                AllFieldTypesDictionary = new() { { "item", new NonNullAllFieldTypesDto() }, { "item2", new NonNullAllFieldTypesDto() } }
            };

        // Act
        var validationResult = await runner.Validate(dto);
        var json = JsonConverter.ToJson(validationResult);

        // Assert
        validationResult.Errors.Should().HaveCount(5);
        Approvals.VerifyJson(json);
    }
}
