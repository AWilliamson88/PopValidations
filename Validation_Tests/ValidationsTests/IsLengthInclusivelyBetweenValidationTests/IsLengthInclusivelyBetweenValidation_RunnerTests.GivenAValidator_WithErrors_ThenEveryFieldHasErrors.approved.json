{
    "errors": {
        "string": [
            "Is not between 0 and 10 inclusive."
        ],
        "allFieldTypesList": [
            "Is not between 0 and 1 inclusive."
        ],
        "allFieldTypesLinkedList": [
            "Is not between 0 and 1 inclusive."
        ],
        "allFieldTypesIEnumerable": [
            "Is not between 0 and 1 inclusive."
        ],
        "allFieldTypesDictionary": [
            "Is not between 0 and 1 inclusive."
        ]
    }
}